src/hooks/useClaims.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from "react-query";
+import { useConnection } from "./useConnection";
+import { ethers } from "ethers";
+import {
+  SUPPORTED_NETWORK_IDS,
+  contracts,
+  currentWindowIndex,
+  URLS,
+} from "../config";
+
+export type Claim = {
+  amount: string;
+  accountIndex: number;
+  metaData: string[];
+  windowIndex: number;
+  proof: string[];
+  ipfsHash: string;
+  rewardToken: string;
+  totalRewardDistributed: string;
+  hasClaimed: boolean;
+};
+export function useClaims() {
+  const { address, network } = useConnection();
+  const { data, isLoading, error, refetch } = useQuery<Claim[], Error>(
+    [address, network?.chainId],
+    () =>
+      getClaims({
+        address: ethers.utils.getAddress(address || ""),
+        chainId: network?.chainId as ChainId,
```

Comment:
nit: to avoid unsafe casts like this, we could ingest the network id in the config (doing runtime verification that it's either 42, 1, or null). Then we could do something like:
```ts
const { chainId } = config(network);
// OR const chainId = config.getChainId(network); if the config logic is a little heavy
... 
      getClaims({
        address: ethers.utils.getAddress(address || ""),
        chainId,
      });
 ...
 ```

knowing the typechecker + runtime verification would ensure that an unexpected value would never make it to this and other functions, and we could probably also drop any passing around of SUPPORTED_NETWORK_IDS.